lib/libvarnish/vss.c

```diff
@@ -100,6 +101,45 @@ vss_parse(char *str, char **addr, char **port)
 	return (NULL);
 }
 
+static const char *
+vss_parse_range(char *str, char **addr, char **port, unsigned long *lo,
+    unsigned long *hi)
+{
+	const char *errp;
+	char *hyphen, *end;
+	unsigned long l, h;
+
+	errp = vss_parse(str, addr, port);
+	if (*port == NULL)
+		return (errp);
+
+	hyphen = strchr(*port, '-');
+	if (hyphen == NULL)
+		return (NULL);
+	if (**port == '-')
+		return ("Range start missing");
```

Comment:
While looking at this PR I found another issue (unrelated to this PR): We never output the error messages, we always get the same generic `"Got no socket(s) for .."` message for any type of failures. We shoud probably output `err` here: https://github.com/varnishcache/varnish-cache/blob/32850ee59cdfd6e35b6ca64367c5f6e05fbac8d2/bin/varnishd/mgt/mgt_acceptor.c#L377

Reply:
Added a commit to include the error reason